### DISCUSSION TITLE
What to do if String.format fails with an exception?

### DISCUSSION THREAD

**Discussion by HanfZero** [B3nedikt/restring#125] [Ideas]
Hi,
we have a question regarding the possible exceptions that might occur when calling String.format:

If the resource strings and the possibly necessary associated format arguments are combined in the ResourcesDelegate#getString or ResourcesDelegate#getQuantityString methods, exceptions of String.format cannot be precluded. This may involve, for example, the number of format arguments or their type. 

Wouldn't it be a good idea to catch the exceptions there and return, for example, the unformatted string from the repository? Wouldn't that be more robust?

André

**Comment by B3nedikt** [comment]
Hi :) While that would be possible, the aim here is to reproduce normal behaviour of the `Resources` class as close as possible, which also doesn't catch any exceptions here. 

I also can't think of any use case for catching exceptions here, I mean you can easily make sure your replaced resources work, just by testing your app. If this is something you need though, you could implement it relatively easily by just checking your strings for the correct number of arguments before adding them to restring, and adding a backup for those were the number of arguments are wrong.

Does this help you?

**Reply by HanfZero** [reply]
Thanks for responding so quickly. :-)

Yes, your response helps to understand the intent here. With that in mind, I wouldn't include exception handling either.

The use case we have to come up with such ideas is that translations can be fetched from the server at runtime. And those translations have the risk that the placeholders may not match the format arguments because of our users' typos, carelessness, and whatnot. :-)
So we don't have the possibility to test that in advance. Hence the idea of exception handling. :-)

But yeah, checking the strings beforehand for correct placeholders might be the easiest solution then. We'll try that.
Well, thanks for the discussion! :-)
